### PR TITLE
Support addRef's initialValue derived from props

### LINKED ISSUE #41: Add support for calculating addRef's initial value based on props
Eg. support using `addRef` like this:

```js
addRef("someRef", props => props.someInitialValue)
```

### PR DESCRIPTION
**Pull request by peterstuart**
Fixes #41.

### CODE REVIEW

File: src/__tests__/addRef.coffee [hunk 1]
```diff
@@ -14,8 +14,32 @@ Comp = flow(
...
+  describe 'with no initial value', ->
+    test 'works', ->
+      {getByText, getByTestId} = render <Comp />
+      fireEvent.click getByText /update/
+      expect(document.activeElement).toBe getByTestId 'input'
+  describe 'with initial value', ->
+    test 'works', ->
+      {getByText} = render <CompWithInitialValue />
+      getByText 'initial'
+      undefined
```
Comment (peterstuart) [inline src/__tests__/addRef.coffee:40]: I'm not sure what your usual approach would be here. (I assume not this.) Calling `getByText` is sufficient as a test, but I need to return `undefined` or a promise.

Comment (helixbass) [reply src/__tests__/addRef.coffee:40]: Ya that's a fine workaround for Coffeescript's implicit return not playing nicely with a library that's finicky about return values

I guess you could override/wrap `test()` to sniff the real `test()`'s return value and only "forward" it if it's a promise

**Review by helixbass — APPROVED** [review]
@peterstuart this looks good, do you mind also updating the README with the updated "type signature" for `addRef()`?

File: src/__tests__/addRef.coffee (hunk 1 above)
Comment (peterstuart) [reply src/__tests__/addRef.coffee:40]: I'm fine with it if it's not too weird.

**Comment by peterstuart** [comment]
@helixbass 👍 README is updated.

**Pull request merged**